[Improve][flink][Connector-V1] Improve druid and doris connector

enterwhat opened the pull request:
https://github.com/apache/incubator-seatunnel/issues/2893

feat #2893 :
(1), solve the problem that the -i parameter cannot be passed in the placeholder of the source part of the configuration file that replaces the conf, and supports placeholder replacement like the sql part.

(2), solve druid's start_date, end_date through the command line -i to pass the parameter space, recognized as the parameter separator of the shell command line, so that the date format parameter of yyyy-MM-dd HH:mm:ss cannot be recognized as a whole, resulting in running Errors, use specific placeholders by default, can be customized

(3), customize the output separator of druid to support importing into doris

(4), replace the default delimiter of doris to solve the problem that " "(\t) in the column value affects column recognition.

(5), druid adds user, password parameter authentication support

(6), provide provided annotation for the package that the druid connector can rely on seatunnel

Commit:
https://github.com/apache/incubator-seatunnel/issues/2893

feat:
(1)，解决-i参数不能传入替换conf的配置文件的source部分占位符的问题，像sql部分一样支持占位符替换。

(2)，解决druid的start_date，end_date通过命令行-i传参空格，识别成shell命令行的参数分隔符导致yyyy-MM-dd HH:mm:ss的日期格式参数不能被整体识别，导致运行出错的问题，默认使用特定占位符，可自定义

(3)，对druid的输出分隔符进行自定义，以便支持导入到doris

(4)，对doris的默认分隔符换成以解决列值里面" "(\t)有影响列识别的问题

(5)，druid添加user，password的参数认证支持

(6)，对druid connector能够依赖seatunnel的包进行provided标注

# Conflicts:
#	seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-druid/pom.xml

EricJoy2048 commented:
Hi, @enterwhat . Thank's for your contribution. Please change the pr comment to english.

enterwhat committed:
https://github.com/apache/incubator-seatunnel/issues/2893

feat #2893 :
(1), solve the problem that the -i parameter cannot be passed in the placeholder of the source part of the configuration file that replaces the conf, and supports placeholder replacement like the sql part.

(2), solve druid's start_date, end_date through the command line -i to pass the parameter space, recognized as the parameter separator of the shell command line, so that the date format parameter of yyyy-MM-dd HH:mm:ss cannot be recognized as a whole, resulting in running Errors, use specific placeholders by default, can be customized

(3), customize the output separator of druid to support importing into doris

(4), replace the default delimiter of doris to solve the problem that " "(\t) in the column value affects column recognition.

(5), druid adds user, password parameter authentication support

(6), provide provided annotation for the package that the druid connector can rely on seatunnel

enterwhat committed:
Merge branch '2.1.3_druid_bugfix_dev' of https://github.com/enterwhat/incubator-seatunnel into 2.1.3_druid_bugfix_dev

enterwhat commented:
> 

done

hailin0 commented on seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/druid/source/DruidSource.java:
```diff
@@ -169,6 +179,8 @@ private RowTypeInfo getRowTypeInfo(String jdbcURL, String datasource, Collection
             i++;
         }
+        long count = Arrays.stream(typeInformation).filter(t -> null == t).count();
+        System.out.println("typeInformation null:::" + count);
```
remove?

enterwhat replied:
done

enterwhat replied:
I

> 

I have commit new commits, should I push a new pr?

EricJoy2048 replied:
> 

I suggest you don't push a new pr if the new pr and this pr have many code dependency. Of course, if there is no code dependency between two prs, you can submit.

enterwhat replied:
> 

any problem with this pr? how long will this be reviewed again?

hailin0 commented on seatunnel-core/seatunnel-core-flink/src/main/java/org/apache/seatunnel/core/flink/command/FlinkApiTaskExecuteCommand.java:
```diff
@@ -63,7 +61,10 @@ public FlinkApiTaskExecuteCommand(FlinkCommandArgs flinkCommandArgs) {
+        this.flinkCommandArgs.getVariables().stream()
+                .map(variable -> variable.split("=", 2))
+                .filter(pair -> pair.length == 2)
+                .forEach(pair -> System.setProperty(pair[0], pair[1]));
```
@EricJoy2048 

please confirm

enterwhat replied:
done

hailin0 commented on seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/druid/source/DruidSource.java:
```diff
@@ -117,17 +117,27 @@ public CheckResult checkConfig() {
+        String endTimestamp = config.hasPath(END_TIMESTAMP) ? config.getString(END_TIMESTAMP).replaceAll(escape_delimiter, " ") : null;
         List<String> columns = config.hasPath(COLUMNS) ? config.getStringList(COLUMNS) : null;
-
+        System.out.println(jdbcURL);
+        System.out.println(user);
+        System.out.println(password);
```
remove?

enterwhat replied:
done

hailin0 commented:
<img width="1105" alt="image" src="https://user-images.githubusercontent.com/14371345/192683622-2ecb594b-66c5-480b-ac5c-428bfc067fca.png">
https://github.com/apache/incubator-seatunnel/actions/runs/3132748547/jobs/5086755036

enterwhat committed:
https://github.com/apache/incubator-seatunnel/issues/2893

feat #2893 :
(1), solve the problem that the -i parameter cannot be passed in the placeholder of the source part of the configuration file that replaces the conf, and supports placeholder replacement like the sql part.

(2), solve druid's start_date, end_date through the command line -i to pass the parameter space, recognized as the parameter separator of the shell command line, so that the date format parameter of yyyy-MM-dd HH:mm:ss cannot be recognized as a whole, resulting in running Errors, use specific placeholders by default, can be customized

(3), customize the output separator of druid to support importing into doris

(4), replace the default delimiter of doris to solve the problem that " "(\t) in the column value affects column recognition.

(5), druid adds user, password parameter authentication support

(6), provide provided annotation for the package that the druid connector can rely on seatunnel

enterwhat committed:
Revert "check config error (#2266)"

This reverts commit a134c0732aec04b85d6894b927c31c0d293c6c5f.

enterwhat committed:
https://github.com/apache/incubator-seatunnel/issues/2893

feat #2893 :
revert code

enterwhat committed:
Merge remote-tracking branch 'origin/dev' into 2.1.3_druid_bugfix_dev

enterwhat committed:
https://github.com/apache/incubator-seatunnel/issues/2893

feat #2893 :
1. Add support for -i parameter in DruidSource
2. The operation of LocalFlinkExample.java is started in jar mode
3. Solve the problem of package conflict error report during the local running of idea

-i start_date=2021-12-0612:00:00 -i end_date=2021-12-0613:00:00  --config "D:\\Users\\Administrator\\Downloads\\apache-seatunnel-incubating-2.1.3\\config\\flink.batch.heartbeat-druid.conf"

enterwhat committed:
Merge branch 'apache:dev' into 2.1.3_druid_bugfix_dev

enterwhat commented:
> Hi, @enterwhat . Thank's for your contribution. Please change the pr comment to english.

I have commit new commits, should I push a new pr?

enterwhat committed:
Merge branch 'apache:dev' into 2.1.3_druid_bugfix_dev

enterwhat committed:
https://github.com/apache/incubator-seatunnel/issues/2893

feat #2893 :
1. fix style check bug

-i start_date=2021-12-0612:00:00 -i end_date=2021-12-0613:00:00  --config "D:\\Users\\Administrator\\Downloads\\apache-seatunnel-incubating-2.1.3\\config\\flink.batch.heartbeat-druid.conf"

enterwhat committed:
Merge branch '2.1.3_druid_bugfix_dev' of https://github.com/enterwhat/incubator-seatunnel into 2.1.3_druid_bugfix_dev

enterwhat committed:
Merge branch 'dev' into 2.1.3_druid_bugfix_dev

enterwhat commented:
> <img alt="image" width="1105" src="https://user-images.githubusercontent.com/14371345/192683622-2ecb594b-66c5-480b-ac5c-428bfc067fca.png">
> 
> https://github.com/apache/incubator-seatunnel/actions/runs/3132748547/jobs/5086755036

I have fixed this

enterwhat commented:
> Hi, @enterwhat . Thank's for your contribution. Please change the pr comment to english.

any problem with this pr? how long will this be reviewed again?

enterwhat committed:
Merge branch 'apache:dev' into 2.1.3_druid_bugfix_dev

CalvinKirs commented:
hi, thanks for your contribution, dev branch will no longer maintain the V1 version of the connector, so I will close this PR,
If you are interested in this, you can contribute the corresponding V2Connector
https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-connectors-v2
